addons/l10n_it_edi/models/account_edi_format.py

```diff
@@ -661,23 +715,18 @@ def _import_fattura_pa(self, tree, invoice):
                             natura_element = element.xpath('.//Natura')
                             invoice_line_form.tax_ids.clear()
                             if percentage is not None:
+                                tax_extra_domain = buyer_seller_info[incoming]['tax_extra_domain']
                                 if natura_element and natura_element[0].text:
                                     l10n_it_kind_exoneration = natura_element[0].text
-                                    tax = self.env['account.tax'].search([
-                                        ('company_id', '=', invoice_form.company_id.id),
-                                        ('amount_type', '=', 'percent'),
-                                        ('type_tax_use', '=', 'purchase'),
-                                        ('amount', '=', percentage),
-                                        ('l10n_it_kind_exoneration', '=', l10n_it_kind_exoneration),
-                                    ], limit=1)
+                                    tax_extra_domain += [('l10n_it_kind_exoneration', '=', l10n_it_kind_exoneration)]
                                 else:
-                                    tax = self.env['account.tax'].search([
-                                        ('company_id', '=', invoice_form.company_id.id),
-                                        ('amount_type', '=', 'percent'),
-                                        ('type_tax_use', '=', 'purchase'),
-                                        ('amount', '=', percentage),
-                                    ], limit=1)
                                     l10n_it_kind_exoneration = ''
```

Comment:
l10n_it_kind_exoneration = natura_element and natura_element[0].text
                                tax = self.env['account.tax'].search([
                                    ('company_id', '=', invoice_form.company_id.id),
                                    ('amount_type', '=', 'percent'),
                                    ('amount', '=', percentage)]
                                    + [('l10n_it_kind_exoneration', '=', l10n_it_kind_exoneration)] if l10n_it_kind_exoneration else []
                                    + type_tax_use_domain,
                                    limit=1
                                )